src/pipeline_migration/actions/add_task.py

```diff
@@ -0,0 +1,406 @@
+            existing_actual_task_names.add(name2)
+
+        if (depended_tasks := task_config.get("runAfter")) is not None:
+            for name in depended_tasks:
+                if name not in existing_pipeline_task_names:
+                    raise ValueError(
+                        f"Task {name} does not exist in the pipeline definition {pipeline_file}."
+                    )
+
+        if pipeline_task_name in existing_pipeline_task_names:
+            logger.info(
+                "Task %s is included in pipeline %s already.", pipeline_task_name, pipeline_file
+            )
+            continue
+
+        if actual_task_name in existing_actual_task_names:
+            logger.info(
+                "Task %s is being referenced in pipeline %s already.",
+                actual_task_name,
+                pipeline_file,
+            )
+            continue
+
+        if (
+            pipeline_task_name in existing_actual_task_names
+            or actual_task_name in existing_pipeline_task_names
+        ):
+            logger.warning(
+                "The pipeline task name and actual task name seem swapped. Skip adding task."
+            )
+            continue
+
+        tasks.append(task_config)
+        dump_yaml(pipeline_file, doc, style)
+        logger.info("Task %s is added to pipeline %s", actual_task_name, original_abs_path)
+
+        if args.git_add:
+            git_add(original_abs_path)
+            logger.info("%s is added to git index.", original_abs_path)
+
+
+def git_add(file_path: FilePath) -> None:
+    """Git add given file
+
+    The git-add command may fail due to any reason, e.g. git command is not available in the system,
+    in which case just logging a message and terminate quietly.
+
+    :param file_path: an absolute path to a file.
+    :type file_path: FilePath
+    :raises ValueError: if given file path is not an absolute path.
+    """
+    fp = Path(file_path)
+    if not fp.is_absolute():
+        raise ValueError(f"File path {file_path} is not an absolute path.")
+    cmd = ["git", "add", fp.name]
+    try:
+        subprocess.run(cmd, cwd=fp.parent, capture_output=True, check=True)
+    except subprocess.CalledProcessError as e:
+        logger.warning("%s is not added to git index: %s", file_path, e.stderr)
+
+
+def search_pipeline_files(files_or_dirs: Iterable[str]) -> Generator[tuple[str, str]]:
+
+    def _iterate_files_or_dirs() -> Generator[Path]:
+        for item in files_or_dirs:
+            if not item:
+                continue
+            entry_path = Path(item).absolute()
+            if entry_path.is_symlink():
+                logger.warning(
+                    "Skip symlink %s. Please specify the destination file or directory instead.",
+                    item,
+                )
+            elif entry_path.is_dir():
+                for entry in entry_path.iterdir():
+                    if entry.is_symlink():
+                        continue
```

Comment:
Why are we skipping symlinks?

Reply:
I didn't figure out a valid use case with symlink. :)

For the validations, this one and the above ones you commented out, I tend to have strict validations first, then relax step by step by specific use cases.